shell/arg.go

```diff
@@ -0,0 +1,110 @@
+package shell
+
+import (
+	"fmt"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+var (
+	escapeNoGlobCharacters = []byte{'|', '&', ';', '<', '>', '(', ')', '$', '`', '\\', '"', '\'', ' ', '\t', '\r', '\n'}
+	// escapeGlobCharacters   = []byte{'*', '?', '['}
+	doubleQuotePattern = regexp.MustCompile(`[^\\][|&;<>()$'" \t\r\n*?[]`)
+)
+
+type ArgType int
+
+const ArgTypeCount = 4
+
+const (
+	ArgConfigEscape       ArgType = iota // escape each special character but don't add quotes
+	ArgConfigNoGlobQuote                 // use double quotes around argument when needed
```

Comment:
The name NoGlob (and "no-glob") made me think at first that glob expressions are forbidden, but in fact it means that glob expressions are passed to restic (as required for include/exclude patterns).

I know this is just internal, but would it make sense to use a name like "preserve-glob" or "keep-glob"?

Reply:
You're right, that helps with code readability 😛